src/hotspot/cpu/x86/vm_version_x86.cpp

```diff
@@ -781,7 +781,7 @@ void VM_Version::get_processor_features() {
               cores_per_cpu(), threads_per_core(),
               cpu_family(), _model, _stepping, os::cpu_microcode_revision());
   assert(res > 0, "not enough temporary space allocated");
-  assert(exact_log2_long(CPU_MAX_FEATURE) + 1 == sizeof(_features_names) / sizeof(char*), "wrong size features_names");
+  assert(log2i_exact((uint64_t)CPU_MAX_FEATURE) + 1 == sizeof(_features_names) / sizeof(char*), "wrong size features_names");
```

Comment:
If you want to get rid of this cast (and the one added to interp_masm_x86.cpp, you could register your log2i functions to accept enums:
-template <typename T, ENABLE_IF(std::is_integral<T>::value)>
+template <typename T, ENABLE_IF(std::is_integral<T>::value || std::is_enum<T>::value)>

Reply:
Good, but feels the extra conditions in the template function is hurting readability more than the single cast - and we should probably apply this more consistently if we go down this route. Mind if I leave this for a follow-up?

I've adressed your other review comments. Thanks for being thorough!